examples/Cargo.toml

```diff
@@ -8,8 +8,9 @@ edition = "2018"
 [features]
 default = ["debug-assertions"]
 metal = ["gfx-backend-metal"]
-gl = ["gfx-backend-gl", "gfx-backend-gl/surfman"]
-wgl = ["gfx-backend-gl", "gfx-backend-gl/wgl"]
+gl = ["gfx-backend-gl", "gfx-backend-gl/surfman"] # GL for Linux
+webgl = ["gfx-backend-gl"] # GL for web
```

Comment:
nit: could we just disable `surfman` on `target_arch = "wasm32"` to avoid the additional `webgl` feature?

Reply:
That was actually how it was initially. I didn't realize it would come into effect here, but while I was combing through the backend implementation I thought that it didn't make any sense to compile with both the wasm32 arch and the `surfman`, `gluten`, or `wgl` features so I turned it into a compile error to combine any of those with `wasm32` arch.

This help a *lot* when it came to the cleanliness of the numerous `#[cfg()]` checks because you didn't have to add in some sort of check for `target_arch = "wasm32"` every time you needed to check for a feature. All you had to do was `feature = "surfman` and know that it was not also going to be targeting `wasm32`.

I tried to get around the extra feature by using cargo's `target` checks but apparently there is either a bug or otherwise strange behavior because if you do the following in the `Cargo.toml`, it adds both the `surfman` *and* the `wgl` feature, even to `wasm32` builds:

```toml
[target.'cfg(target_arch = "wasm32")'.dependencies.gfx-backend-gl]
path = "../src/backend/gfx-backend-gl"
optional = true

[target.'cfg(windows)'.dependencies.gfx-backend-gl]
path = "../src/backend/gfx-backend-gl"
features = ["wgl"]
optional = true

[target.'cfg(unix)'.dependencies.gfx-backend-gl]
path = "../src/backend/gfx-backend-gl"
features = ["surfman"]
optional = true
```

Its unfortunate that that doesn't work because that would allow us to automatically detect the correct feature without even having to have a separate one for `wgl`.

Anyway, I think that the extra feature is worth the readability in the code, especially because we can't get away from having separate features for `wgl` and `gl` anyway.

Reply:
Ok sounds good to me :+1:

In general I think having the extra `#cfg`s in our own implementation (even at the cost of some readability) is usually preferable to exposing additional features – mostly to make it easier for users to use. But there's not much we can do here if cargo doesn't easily support this :)

Reply:
Yeah, it would be *perfect* if cargo supported this. I opened a [forum topic](https://users.rust-lang.org/t/cargo-confusion-with-feature-and-target-specific-dependencies/40083?u=zicklag) just now to see if we can get any feedback on whether or not that could work.

Maybe its just a Cargo bug that we could fix, or maybe there is a workaround.

Reply:
This is a known issue in cargo: https://github.com/rust-lang/cargo/issues/1197

A solution seems to be available in nightly: https://github.com/rust-lang/cargo/issues/7914

Reply:
Cool, at least that means we'll be able to make that nicer after that gets stabilized. I'll subscribe to that thread and once it gets merged we could fix it if we don't mind requiring using the latest stable Rust.